image-builder/image_builder.sh

```diff
@@ -392,6 +392,7 @@ create_rootfs_image() {
 	touch "${mount_dir}/etc/machine-id"
 
 	info "Creating empty resolv.conf to allow kata-agent to bind-mount it"
+	rm -f "${mount_dir}/etc/resolv.conf"
```

Comment:
Why rm'ing the existing /tmp/resolv.conf? If the goal is to bind mount, you can keep the original file.

Reply:
@marcov: Incase of Ubuntu based rootfs, the existing /etc/resolv.conf is symlinked to "/run/systemd/resolve/stub-resolv.conf" which is a valid file when os-builder creats the rootfs and creates an image out of it. But when Kata VM is spun up and we 9pfs mount kata files on top of /run, that resolv.conf file under /run is inaccessible and hence agent mount fails with no such file or directory.

Reply:
Thanks! Then maybe you can check if the file is a symlink with `[ -L ` ... and just delete it in that case. This way we would support custom rootfs created by the user.
Also, why dont do rm & touch in the rootfs-builder.sh script?

Reply:
@marcov : makes sense. Updated PR. Please take a look.